arctic_training/data/factory.py

```diff
@@ -79,21 +80,25 @@ def __init__(self, trainer: "Trainer", config: Optional[DataConfig] = None) -> N
     def __call__(self) -> Tuple[DataLoader, Optional[Mapping[str, DataLoader]]]:
         def get_data_split(split: str) -> Optional[DatasetType]:
             data_sources = self._get_data_sources(split=split)
-
-            cache_path = self.cache_path(sources=data_sources, split=split)
-            if self.config.use_data_cache and cache_path.exists():
-                logger.info(f"Loading dataset from cache path {cache_path.as_posix()}")
-                return load_from_disk(cache_path.as_posix())
-
             if len(data_sources) == 0:
                 return None
-            dataset = self.load(data_sources, split=split)
-            dataset = self._truncate_data(dataset)
 
-            if self.config.use_data_cache:
+            cache_path = self.cache_path(sources=data_sources, split=split)
+
+            # If the cache path does not exist, load the data using local/global
+            # rank 0 (depending on if file system is shared across nodes).
+            if self.is_main_process_by_path(cache_path) and not cache_path.exists():
+                dataset = self.load(data_sources, split=split)
                 logger.info(f"Saving dataset to cache path {cache_path.as_posix()}")
                 dataset.save_to_disk(cache_path.as_posix())
-            return dataset
+
+            dist.barrier()  # Wait for main process to finish saving to cache
+
+            self.trainer._set_seeds(
+                self.trainer.config.seed
+            )  # Reset seeds before processing data
```

Comment:
```suggestion
            # Reset seeds before processing data
            self.trainer._set_seeds( self.trainer.config.seed)
```
no need to create new lines and reduce readability - the comment after only works functionality if autoformatter doesn't wrap it. most likely you just didn't expect it to get wrapper - let's switch to wider width soon!

Reply:
also I don't think the comment is right, shouldn't it say "after processing data if at all"

but also perhaps add a longer comment to suggest why we are doing this? perhaps:
```
# Reset seeds after may be processing data if cache didn't exist - so that main process ends up with the same RNG if the cache was there and if it wasn't.
```